### PR TITLE
fix(compiler): Enable Windows Support by Default

### PR DESCRIPTION
**Pull request by sgoudham**
- [x] Match 'Windows_NT' uname
- [x] Ensure that the default cache path is suitable for Windows users

### CODE REVIEW

**Comment by Pocco81** [comment]
I'm not on Windows so I really can't be sure if this works. Code looks good overall.

Have you tested it yet? Did it work?

File: lua/catppuccin/lib/compiler.lua [hunk 1]
```diff
@@ -2,6 +2,8 @@ local M = {}
 
 -- Credit: https://github.com/EdenEast/nightfox.nvim
 local fmt = string.format
+local sysname = vim.loop.os_uname().sysname
```
Comment (nullchilly) [inline lua/catppuccin/lib/compiler.lua:5]: I think this is better:
```lua
local is_windows = vim.startswith(vim.loop.os_uname().sysname, "Windows")
```

File: lua/catppuccin/lib/compiler.lua [hunk 2]
```diff
@@ -68,11 +74,15 @@ vim.g.colors_name = "catppuccin"]],
 		end
 	end
 	os.execute(
-		string.format("mkdir %s %s", vim.loop.os_uname().sysname == "Windows" and "" or "-p", config.compile.path)
+		string.format(
+			"mkdir %s %s",
+			(sysname == "Windows" or sysname == "Windows_NT") and "" or "-p",
```
Comment (nullchilly) [inline lua/catppuccin/lib/compiler.lua:79]:
```lua
is_windows and "" or "-p"
```
Seems much cleaner IMO

**Review by nullchilly — CHANGES_REQUESTED** [review]

**Review by nullchilly — APPROVED** [review]

**Comment by nullchilly** [comment]
Can confirm it works on windows 👍🏻 
```
032.111  000.173  000.173: require('catppuccin')
032.431  000.307  000.307: require('catppuccin.config')
034.670  000.042  000.042: require('vim.F')
038.035  002.350  002.350: sourcing C:\Users\nullchilly\AppData\Local\nvim-data\site\pack\packer\start\theme\colors\catppuccin.lua
```

**Pull request merged**